cpp/src/arrow/type_traits.cc

```diff
@@ -86,9 +86,13 @@ int RequiredValueAlignmentForBuffer(Type::type type_id, int buffer_index) {
     case Type::DICTIONARY:
     case Type::EXTENSION:
     case Type::MAX_ID:
-      Status::Invalid("Could not check alignment for type id ", type_id).Warn();
+      Status::Invalid("RequiredValueAlignmentForBuffer called with invalid type id ",
+                      type_id)
+          .Warn();
```

Comment:
You could put a `break` here and move this message to where you added the other status before the `return 1` at the end. Both situations are only possible with invalid arguments being passed anyways.

Reply:
Done.  I was scratching my head trying to figure out how to unify these.  Thanks.